src/Middleware/ProviderMiddleware.php

```diff
@@ -58,12 +58,18 @@ public function handle(Request $request, Closure $next)
         $provider = $this->providerResolver->getProvider($user);
 
         // add provider to symfony request
-        $request->attributes->add([
-            'provider' => $provider
-        ]);
+        $request->attributes->add(
+            \array_merge($route[2] ?? [], [
+                'provider' => $provider
+            ])
+        );
 
         // add provider to lumen route
-        $route = \array_merge($route, [['provider' => $provider]]);
+        // Put the Symfony request attributes back into the laravel route.
```

Comment:
You dont need to do this one. The only reason the param converter middleware does it is because that code operates on the symfony attribute bag.

You also only need to add the specific 'provider' key to the symfony attributebag, not the entire route.